Sources/PackageGraph/Pubgrub/PubgrubDependencyResolver.swift

```diff
@@ -21,7 +21,7 @@ public struct PubgrubDependencyResolver {
     public typealias Constraint = PackageContainerConstraint
 
     /// the mutable state that get computed
-    internal final class State {
+    public final class State {
```

Comment:
I'd prefer not making this public since it mutable state and changing it from the outside may cause serious issues

Reply:
Agreed.  I only did that because of a compiler error since it's being used in the case of a public enum.  I'll see if I can find another way to do this.